src/standalone/Selector/MultiSelectWithTags.tsx

```diff
@@ -0,0 +1,302 @@
+import React, { useState, useCallback, useRef } from "react";
+import { SelectorData, SelectorPropsSingleSelect } from "./Selector";
+import SingleSelect from "./Selector";
+import { TextField, TextFieldProps, Tooltip } from "@material-ui/core";
+import { createStyles, Theme, withStyles, WithStyles } from "@material-ui/core";
+import { SmallIconButton, SmallListItemIcon } from "../Small/List";
+import { Search as SearchIcon, Info as InfoIcon } from "@material-ui/icons";
+import { RemoveIcon } from "../../standalone";
+import { Autocomplete } from "@material-ui/lab";
+import { GenericWithStyles } from "../../utils";
+
+export interface MultiSelectData extends SelectorData {
+	/**
+	 * Item type value
+	 */
+	type?: string;
+	/**
+	 * Item click handler
+	 */
+	onClick?: () => void;
+	/**
+	 * Can the entry be unselected?
+	 * @param data The data entry to be unselected
+	 */
+	canUnselect?: (data: MultiSelectData) => boolean | Promise<boolean>;
+}
+
+export interface MultiSelectWithTagsProps<Data extends MultiSelectData>
+	extends Omit<
+		SelectorPropsSingleSelect<Data>,
+		"multiSelect" | "clearable" | "onSelect" | "selected"
+	> {
+	/**
+	 * Extended selection change handler
+	 * @param data The selected data entries
+	 */
+	onSelect?: (value: Data[]) => void;
+	/**
+	 * Extended group selection change handler
+	 * @param data The selected data entry
+	 */
+	onGroupSelect?: (value: SelectorData | null) => void;
+	/**
+	 * The title of control
+	 */
+	title: string;
+	/**
+	 * The currently selected values
+	 */
+	selected: Data[];
+	/**
+	 * The filtered data after selected dropdown value
+	 */
+	filteredData: Data[];
+	/**
+	 * Set filtered data after selected from search input
```

Comment:
Clearify please, is this some kind of onChange event for `filteredData`? If yes please include it in the comment

Reply:
yes, it's callback function so comment updated